File: anagram.pl
#!/usr/bin/env perl

open(DICT, "words") || die "Can't open the dictionary";

while(<DICT>)
	{
	chomp($word = $_);
	push(@{$wordHash{join('', sort split(//, $word))}}, $word);
	}

close(DICT);

foreach $key (sort keys %wordHash)
	{
	if($#{$wordHash{$key}} > 2)
		{
		print "$key\t=>\t", join(', ', @{$wordHash{$key}}), "\n";
		}
	}

File: anagram.rb
#!/usr/bin/env ruby

wordHash = Hash.new()

File.new("words").each do |word|
	word.chomp!
	key = word.split(//).sort.join('')
	wordHash[key] = Array.new unless(wordHash.has_key?(key))
	wordHash[key] << word
end

wordHash.sort.each {|k,v| puts "#{k}\t=>\t#{v.join(', ')}" if(v.size > 3) }
